src/routes/mod_creation.rs

```diff
@@ -0,0 +1,228 @@
+use crate::file_hosting::{upload_file, AuthorizationData, UploadUrlData};
+use crate::models::ids::random_base62;
+use crate::models::mods::{
+    FileHash, GameVersion, Mod, ModId, Version, VersionFile, VersionId, VersionType,
+};
+use crate::models::teams::{Team, TeamId, TeamMember};
+use actix_multipart::{Field, Multipart};
+use actix_web::web::Data;
+use actix_web::{middleware, post, HttpResponse};
+use bson::Bson;
+use chrono::Utc;
+use futures::StreamExt;
+use mongodb::Client;
+use pulldown_cmark::html::push_html;
+use pulldown_cmark::{Options, Parser};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone)]
+struct InitialVersionData {
+    pub file_names: Vec<String>,
+    pub version_slug: String,
+    pub version_title: String,
+    pub version_body: String,
+    pub dependencies: Vec<ModId>,
+    pub game_versions: Vec<GameVersion>,
+    pub version_type: VersionType,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+struct ModCreateData {
+    /// The title or name of the mod.
+    pub mod_name: String,
+    /// The namespace of the mod
+    pub mod_namespace: String,
+    /// A short description of the mod.
+    pub mod_description: String,
+    /// A long description of the mod, in markdown.
+    pub mod_body: String,
+    /// A list of initial versions to upload with the created mod
+    pub initial_versions: Vec<InitialVersionData>,
+    /// The team of people that has ownership of this mod.
+    pub team_members: Vec<TeamMember>,
+    /// A list of the categories that the mod is in.
+    pub categories: Vec<String>,
+    /// An optional link to where to submit bugs or issues with the mod.
+    pub issues_url: Option<String>,
+    /// An optional link to the source code for the mod.
+    pub source_url: Option<String>,
+    /// An optional link to the mod's wiki page or other relevant information.
+    pub wiki_url: Option<String>,
+}
+
+#[post("api/v1/mod")]
```

Comment:
As said above, move the logic to the routes module

Reply:
That method is currently in the `routes::mod_creation` module.